src/bsoncxx/types/value.cpp

```diff
@@ -56,35 +74,52 @@ value& value::operator=(const value& rhs) {
 
 value::value(value&& rhs) noexcept {
     *this = std::move(rhs);
+    rhs._type = stdx::nullopt;
```

Comment:
I checked, and moving out of an optional does not make the rhs a nullopt. I would expect it to also.